### DISCUSSION TITLE
How to get a Shopid

### DISCUSSION THREAD

**Discussion by RichMitchell** [etsy/open-api#1242] [Q&A]
Lots of the API calls require a ShopId be provided as art of the request URL; we have multiple customers who use our Commerical App to manage their Etsy order fulfillment, but they need to provide their ShopId into our account objects so they can perform a getShopReceipts call.  
How is an Etsy user meant to find their generic ShopId other than implementing a call to findShops and searching for it via the ShopName?  
Is there anywhere in the front end Shop Management that a Etsy customer can look to find their ShopId?

**Comment by mjec** [comment]
Take a look at the [`getMe`](https://developers.etsy.com/documentation/reference#operation/getMe) endpoint, which provides you with the `user_id` and `shop_id` for the token used to call it.

**Reply by alexlyeh** [reply]
Do you know if there is a way to get the shop_id without having to go through oauth?
